src/o365/spo/commands/hubsite/hubsite-theme-sync.ts

```diff
@@ -0,0 +1,130 @@
+import auth from '../../SpoAuth';
+import config from '../../../../config';
+import * as request from 'request-promise-native';
+import commands from '../../commands';
+import {
+  CommandOption, CommandValidate
+} from '../../../../Command';
+import SpoCommand from '../../SpoCommand';
+import Utils from '../../../../Utils';
+import GlobalOptions from '../../../../GlobalOptions';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  webUrl: string;
+}
+
+class SpoHubSiteThemeSyncCommand extends SpoCommand {
+  public get name(): string {
+    return `${commands.HUBSITE_THEME_SYNC}`;
+  }
+
+  public get description(): string {
+    return 'Applies any theme updates from the parent hub site.';
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    auth
+      .ensureAccessToken(auth.service.resource, cmd, this.debug)
```

Comment:
This should be `auth.getAccessToken` instead, because on line 40 you're issuing the request on the URL of the site specified by the user. If the CLI is connected to SPO on the tenant admin site (`spo connect https://contoso-admin.sharepoint.com`) and you try to apply theme updates for a regular site, eg. `spo hubsite theme sync -u https://contoso.sharepoint.com/sites/marketing`, the request fails because the site is located on a different domain and requires a different access token. We have more info on this topic in our docs at https://sharepoint.github.io/office365-cli/concepts/authorization-tokens/.